airflow/providers/microsoft/azure/hooks/wasb.py

```diff
@@ -172,7 +172,7 @@ def _get_blob_client(self, container_name: str, blob_name: str) -> BlobClient:
         :param blob_name: The name of the blob. This needs not be existing
         :type blob_name: str
         """
-        container_client = self.create_container(container_name)
+        container_client = self._get_container_client(container_name)
```

Comment:
The current behaviour is intentional. We create the container if it doesn't already exist. If it already exists, we return the client of the already existing container.  However, we can improve it by having `_get_blob_client` take an optional argument like `create_container_if_not_existing=True`. If True we call `create_container` else we call `_get_container_client`

Reply:
> The current behaviour is intentional. We create the container if it doesn't already exist. If it already exists, we return the client of the already existing container. However, we can improve it by having `_get_blob_client` take an optional argument like `create_container_if_not_existing=True`. If True we call `create_container` else we call `_get_container_client`

The behaviour is quite unexpected. E.g. if I want to check for a blob I do not want a container to be created, because I am expecting a blob in an existing container. Furthermore, if connecting with read only permission, the current code throws an exception of permission denied.

Reply:
For backward compatibility, what do you think about using try/except here? e.g:
```python
try:
    container_client = self.create_container(container_name)
except PermissionError:
    container_client = self._get_container_client(container_name)
```
We can find the actual error to catch.

Reply:
One of the options, yes. 
But in general, do you think it is correct to do a `write` operation during an action that is `read-only`? An example would be a blob sensor. 
Strangely, the prefix sensor is not using `_get_blob_client` thus it works correctly without `rw` permissions.

Reply:
We can actually make this a breaking change but have an argument on the method instead of try/except. Not Ok to do a write operation when you only have a` read-only` perms. So I will go for:
```python
def _get_blob_client(self, container_name: str, blob_name: str, existing_container=True):
    if existing_container:
        container_client = self._get_container_client(container_name)
   else:
        container_client = self.create_container(container_name)
...
```
We have many methods that depend on this, so changing it would mean a breaking change but @potiuk can advise the best option here

Reply:
Instead of the above, we can edit the `create_container` method to return the container client if the container exists before creating a container. That way it won't be a breaking change. Can you try that @ignaski

Reply:
`_get_blob_client` is used in 3 methods: `check_for_blob`, `download` and `upload`. Only the `upload` is `write` type of operation. In this case, would it make sense to call `create_container` in the `upload` before asking for blob client? This way the method `_get_blob_client` could only be responsible for returning the client.